solutions/src/main/java/ru/sberbank/school/task13/cache/CacheProxy.java

```diff
@@ -0,0 +1,334 @@
+
+                        try (InputStream in = zipFile.getInputStream(zipEntry)) {
+                            byte[] bytes = getBytes(in);
+
+                            try (ObjectInputStream ois = new ObjectInputStream(new ByteArrayInputStream(bytes))) {
+                                result = ois.readObject();
+                            }
+                        }
+                    }
+                }
+            } catch (IOException | ClassNotFoundException e) {
+                throw new CacheProxyException("Error occurred while loading file: " + file.getName() +
+                        "from a zip", e);
+            }
+
+            return result;
+        }
+
+        private byte[] getBytes(InputStream in) throws IOException {
+            byte[] buffer = new byte[in.available()];
+            try (ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
+                int r;
+
+                while ((r = in.read(buffer)) != -1) {
+                    baos.write(buffer,0, r);
+                }
+
+                return baos.toByteArray();
+            }
+        }
+
+        private void writeToCacheFile(File file, Object toSave) {
+           try {
+               isSerializableCheck(toSave);
+
+               if (isZip(file.getName())) {
+                   writeZipFile(file, toSave);
+               } else {
+                   writeToFile(file, toSave);
+               }
+
+            } catch (IOException e) {
+                throw new CacheProxyException("Error occurred while writing object: " +
+                        toSave + " to file: " + file.getName(), e);
+            }
+        }
+
+        private void writeZipFile(File file, Object toSave) throws IOException {
+            isSerializableCheck(toSave);
+
+            File zipFile = new File(file.getAbsolutePath() + ".zip");
+            try (ZipOutputStream zout = new ZipOutputStream(new FileOutputStream(zipFile));
+                 ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                 ObjectOutputStream oos = new ObjectOutputStream(baos)) {
+
+                ZipEntry zipEntry = new ZipEntry(file.getName());
+                zout.putNextEntry(zipEntry);
+                oos.writeObject(toSave);
+                zout.write(baos.toByteArray());
+
+                zout.closeEntry();
+            }
+        }
+
+        private void writeToFile(File file, Object toSave) throws IOException {
+            try (FileOutputStream fout = new FileOutputStream(file);
+                ObjectOutputStream out = new ObjectOutputStream(fout)) {
+                out.writeObject(toSave);
+            }
+        }
+
+        private String getKeyToCache(Method method, Object[] args) {
+            Cache annotation = method.getAnnotation(Cache.class);
+            StringBuilder key = new StringBuilder();
+            boolean isZip = annotation.zip();
+            String methodName = !annotation.serviceName().isEmpty() ?
```

Comment:
&#39;?&#39; should be on a new line.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1473056551>*